src/CustomModal/CustomModal.js

```diff
@@ -0,0 +1,143 @@
+    /** Passed to the secondary button as props without any filter / mutation */
+    secondaryButtonProps: PropTypes.object,
+    /** callback to be called when the secondary button is clicked */
+    secondaryButtonOnClick: PropTypes.func,
+    /** The modal's title */
+    title: PropTypes.string,
+    /** The modal's subtitle */
+    subtitle: PropTypes.string,
+    /** When set to true, there will be no content padding */
+    removeContentPadding: PropTypes.bool,
+    /** a footnote node, to be rendered at the very bottom of the modal */
+    footnote: PropTypes.node,
+    /** side actions node, to be rendered as the first element on the same row as the action buttons */
+    sideActions: PropTypes.node,
+    /** the children / content of the modal */
+    children: PropTypes.any,
+  };
+
+  static defaultProps = {
+    dataHook: dataHooks.customModal,
+    primaryButtonText: 'Confirm',
+    primaryButtonProps: { dataHook: dataHooks.primaryButton },
+    secondaryButtonText: 'Cancel',
+    secondaryButtonProps: { dataHook: dataHooks.secondaryButton },
+    title: '',
+    subtitle: '',
+    removeContentPadding: false,
+  };
+
+  _renderFooterLayout = () => {
+    const {
+      sideActions,
+      secondaryButtonText,
+      secondaryButtonOnClick,
+      secondaryButtonProps,
+      primaryButtonText,
+      primaryButtonOnClick,
+      primaryButtonProps,
+    } = this.props;
+    return (
+      <div>
+        <Divider className={styles.footer_divider} />
+        <div className={styles.footer}>
+          {sideActions && (
+            <div className={styles.sideActions}>{sideActions}</div>
+          )}
+          <div className={styles.buttons}>
+            {secondaryButtonText && (
+              <Button
+                dataHook={dataHooks.secondaryButton}
+                onClick={secondaryButtonOnClick}
+                priority="secondary"
+                size="small"
+                {...secondaryButtonProps}
+              >
+                {secondaryButtonText}
+              </Button>
+            )}
+            {primaryButtonText && (
+              <Button
+                dataHook={dataHooks.primaryButton}
+                onClick={primaryButtonOnClick}
+                size="small"
+                {...primaryButtonProps}
+              >
+                {primaryButtonText}
+              </Button>
+            )}
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  render() {
+    const {
+      dataHook,
+      primaryButtonText,
+      secondaryButtonText,
+      title,
+      subtitle,
+      removeContentPadding,
+      children,
+      sideActions,
+      footnote,
+      onClose,
+    } = this.props;
+
+    const classes =
+      'customModal' + (removeContentPadding ? ' removeContentPadding' : '');
+    const hasFooter = sideActions || primaryButtonText || secondaryButtonText;
+
+    return (
+      <div {...styles(classes, {}, this.props)} data-hook={dataHook}>
```

Comment:
Let's do it in the Stylable way:
```suggestion
      <div {...styles('root', { removeContentPadding }, this.props)} data-hook={dataHook}>
```

And then you can use the state in your `.st` file, like:
https://github.com/wix/wix-style-react/blob/master/src/SidebarSectionItem/SidebarSectionItem.st.css#L30

Reply:
Done